Savanna/Model/Animal.cs

```diff
@@ -0,0 +1,196 @@
+        protected static List<AnimalNewCoordinates> FreeCellsToMove(List<Animal> animals, Animal animal)
+        {
+            List<AnimalNewCoordinates> newAnimalsCoordinates = new();
+            AnimalNewCoordinates newCoordinates;
+
+            for (int newYCoordinate = animal.YCoordinate - 1; newYCoordinate <= animal.YCoordinate + 1; newYCoordinate++)
+            {
+                for (int newXCoordinate = animal.XCoordinate - 1; newXCoordinate <= animal.XCoordinate + 1; newXCoordinate++)
+                {
+                    if (!IsCellReserved(newYCoordinate, newXCoordinate, animals) && !Game.IsCellOnBoard(newYCoordinate, newXCoordinate))
+                    {
+                        newCoordinates = new AnimalNewCoordinates
+                        {
+                            NewYCoordinate = newYCoordinate,
+                            NewXCoordinate = newXCoordinate
+                        };
+
+                        newAnimalsCoordinates.Add(newCoordinates);
+                    }
+                }
+            }
+
+            newCoordinates = new AnimalNewCoordinates
+            {
+                NewYCoordinate = animal.YCoordinate,
+                NewXCoordinate = animal.XCoordinate
+            };
+
+            newAnimalsCoordinates.Add(newCoordinates);
+            return newAnimalsCoordinates;
+        }
+
+        /// <summary>
+        /// Checks if cell is reserved.
+        /// </summary>
+        /// <param name="newXCoordinate">New X position.</param>
+        /// <param name="newYCoordinate">New Y position.</param>
+        /// <param name="animals">List of animals.</param>
+        /// <returns>Is cell reserved.</returns>
+        private static bool IsCellReserved(int newXCoordinate, int newYCoordinate, List<Animal> animals)
+        {
+            for (int element = 0; element < animals.Count; element++)
+            {
+                Animal? animal = animals[element];
+                if (animal.XCoordinate.Equals(newXCoordinate) && animal.YCoordinate.Equals(newYCoordinate))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Looks around game field from animals based on it vision.
+        /// </summary>
+        /// <param name="animal">Current animal.</param>
+        /// <param name="animals">List of animals.</param>
+        /// <param name="vision">How many cells near animal can sense other animals.</param>
+        /// <returns>Enumerable list with nearby animals.</returns>
+        public static IEnumerable<Animal> LookAround(Animal animal, IEnumerable<Animal> animals, int vision)
+        {
+            return from animalForSearch in animals
+                   where AnimalsAreInRange(animal, animalForSearch, vision)
+                   select animalForSearch;
+        }
+
+        /// <summary>
+        /// Returns all animals in the range of vision, excluding itself.
```

Comment:
Return type is bool. So comment is wrong.

Reply:
Comment changed.